x-pack/plugins/global_search_bar/public/lib/result_to_option.test.ts

```diff
@@ -115,24 +115,20 @@ describe('resultToOption', () => {
       meta: { categoryLabel: 'category', displayName: 'foo', tagIds: ['known', 'unknown'] },
     });
 
-    const getTag = (tagId: string): Tag | undefined => {
-      if (tagId === 'known') {
-        return {
+    const getTagList = (): Tag[] => {
+      return [
+        {
           id: 'known',
           name: 'Known',
           description: 'Known',
           managed: false,
           color: '#000000',
-        };
-      }
+        },
+      ];
     };
 
-    const logSpy = jest.spyOn(console, 'warn').mockImplementation();
+    const option = resultToOption(input, [], getTagList);
 
-    const option = resultToOption(input, [], getTag);
-    expect(logSpy).toBeCalledWith(
-      'SearchBar: Tag with id "unknown" not found. Tag "unknown" is referenced by the search result "dashboard:id". Skipping displaying the missing tag.'
-    );
```

Comment:
The `console.warn` in `result_to_option.ts` will never be called, because doing `const tags = getTagList().filter` loops through the known tags. I think we should keep the logic a little bit more similar to the old functionality by looping through the tagIds from `result.meta` - some of the tags in the result could be unknown and I'd like to preserve that warning log for that scenario.

Making that change in `result_to_option.ts` would cause fewer subtractions in this test file and we can keep the `expect(logSpy).toBeCalledWith(...)` part.